deploy/bz1980755/10-bz1980755.CronJob.yaml

```diff
@@ -40,35 +40,46 @@ spec:
+              fi
 
-                      echo "Will re-install subscription ${ns}/${sub}"
-                      oc get subscriptions.operators.coreos.com -n "${ns}" "${sub}" -ojson | grep -v "kubectl.kubernetes.io/last-applied-configuration" > ${TEMPDIR}/${sub}.yaml
+              while read subAndNS; do
 
-                      for ip in `oc get installplans.operators.coreos.com -n "${ns}" -o jsonpath="{.items[?(@.metadata.ownerReferences[*].name=='${sub}')].metadata.name}"`; do
-                          echo "Removing installplan ${ip}"
-                          oc delete installplans.operators.coreos.com -n "${ns}" $ip
-                      done
+                  ns=$(echo "${subAndNS}" | cut -d, -f 1)
+                  sub=$(echo "${subAndNS}" | cut -d, -f 2)
 
-                      for csv in `oc get clusterserviceversions.operators.coreos.com -n "${ns}" -o jsonpath="{.items[*].metadata.name}"`; do
-                          if [[ "${csv}" =~ ^${sub}.* || "openshift-${csv}" =~ ^${sub}.* ]]; then
-                              echo "Removing CSV ${csv}"
-                              oc delete clusterserviceversions.operators.coreos.com -n "${ns}" "${csv}"
-                          fi
-                      done
+                  if [[ -z "${ns}" || -z "${sub}" ]]; then
+                      continue
+                  fi
 
-                      echo "Removing subscription ${sub}"
-                      oc delete subscriptions.operators.coreos.com  -n "${ns}" "${sub}"
+                  echo "Will re-install subscription ${ns}/${sub}"
+                  oc get subscriptions.operators.coreos.com -n "${ns}" "${sub}" -ojson | grep -v "kubectl.kubernetes.io/last-applied-configuration" > ${TEMPDIR}/${sub}.yaml
 
-                      echo "Recreating subscription ${sub}"
-                      oc create -f ${TEMPDIR}/${sub}.yaml
+                  for ip in `oc get installplans.operators.coreos.com -n "${ns}" -o jsonpath="{.items[?(@.metadata.ownerReferences[*].name=='${sub}')].metadata.name}"`; do
+                      echo "Removing installplan ${ip}"
+                      oc delete installplans.operators.coreos.com -n "${ns}" $ip
+                  done
+
+                  for csv in `oc get clusterserviceversions.operators.coreos.com -n "${ns}" -o jsonpath="{.items[*].metadata.name}"`; do
+                      if [[ "${csv}" =~ ^${sub}.* || "openshift-${csv}" =~ ^${sub}.* ]]; then
+                          echo "Removing CSV ${csv}"
+                          oc delete clusterserviceversions.operators.coreos.com -n "${ns}" "${csv}"
+                      fi
                   done
-              done
+
+                  echo "Removing subscription ${sub}"
+                  oc delete subscriptions.operators.coreos.com  -n "${ns}" "${sub}"
+
+                  echo "Restarting catalog-operator to clear cache"
+                  oc delete pod -n openshift-operator-lifecycle-manager -l app=catalog-operator --wait=true
```

Comment:
```
Restarting catalog-operator to clear cache
Error from server (Forbidden): pods is forbidden: User "system:serviceaccount:openshift-sre-pruning:bz1980755" cannot list resource "pods" in API group "" in the namespace "openshift-operator-lifecycle-manager"
```

I am seeing error when testing on my staging cluster. Seems the clusterrole is missing the permissions for the pods?

Reply:
Nice, thanks for finding and spotting that! The `ClusterRole` did indeed require additional permissions. I've updated it to reflect that in the PR now.